web/core/src/jsMain/kotlin/org/jetbrains/compose/web/css/StyleSheet.kt

```diff
@@ -79,49 +78,32 @@ open class StyleSheet(
 
     @Suppress("EqualsOrHashCode")
     internal class CSSSelfSelector(var selector: CSSSelector? = null) : CSSSelector() {
-        override fun toString(): String = throw IllegalStateException("You can't concatenate `String + CSSSelector` which contains `self` or `root`. Use `selector(<your string>)` to convert `String` to `CSSSelector` for proper work. https://github.com/JetBrains/compose-jb/issues/1440")
-        override fun asString(): String = selector?.asString() ?: throw IllegalStateException("You can't instantiate self")
+        override fun toString(): String =
+            throw IllegalStateException("You can't concatenate `String + CSSSelector` which contains `self` or `root`. Use `selector(<your string>)` to convert `String` to `CSSSelector` for proper work. https://github.com/JetBrains/compose-jb/issues/1440")
+
+        override fun asString(): String =
+            selector?.asString() ?: throw IllegalStateException("You can't instantiate self")
+
         override fun equals(other: Any?): Boolean {
             return other is CSSSelfSelector
         }
     }
 
-    // TODO: just proof of concept, do not use it
-    @ExperimentalComposeWebStyleApi
-    fun css(cssBuild: CSSBuilder.() -> Unit): String {
-        val selfSelector = CSSSelfSelector()
-        val (style, newCssRules) = buildCSS(selfSelector, selfSelector, cssBuild)
-        val cssRule = cssRules.find {
-            it is CSSStyleRuleDeclaration &&
-                    it.selector is CSSSelector.CSSClass && it.style == style &&
-                    (boundClasses[it.selector.className] ?: emptyList()) == newCssRules
-        }.unsafeCast<CSSStyleRuleDeclaration?>()
-        return if (cssRule != null) {
-            cssRule.selector.unsafeCast<CSSSelector.CSSClass>().className
-        } else {
-            val classNameSelector = CSSSelector.CSSClass("auto-${counter++}")
-            selfSelector.selector = classNameSelector
-            add(classNameSelector, style)
-            newCssRules.forEach { add(it) }
-            boundClasses[classNameSelector.className] = newCssRules
-            classNameSelector.className
-        }
-    }
-
     protected class CSSHolder(private val usePrefix: Boolean, private val cssBuilder: CSSBuilder.() -> Unit) {
         operator fun provideDelegate(
             sheet: StyleSheet,
             property: KProperty<*>
         ): ReadOnlyProperty<Any?, String> {
             val sheetName = if (usePrefix) "${sheet::class.simpleName}-" else ""
-            val selector = CSSSelector.CSSClass("$sheetName${property.name}")
```

Comment:
was it incorrect? why can't we use `CSSSelector.CSSClass` here?

Reply:
I see now that it was removed. Can method `className ` be used then?

Reply:
className is unfortunately a context method but actually I see your point - let me think what can be done.